### PR TITLE
chore(storybook): add pages for chromatic theme snapshots

### PR DESCRIPTION
**Pull request by williamernest**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Adds support for chromatic snapshots in each theme (ng-light, ng-dark, core-light, core-dark, high-contrast). 

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .storybook/helpers/setup-storybook.helpers.ts [hunk 1]
```diff
@@ -32,13 +34,31 @@ export function setupStorybook(
     );
 
   storyApi.add('Default', defaultStory, defaultParameters);
-
-  storyApi.add('Variants', variants.length ? combineStories(defaultStory, variants) : defaultStory, {
+  const combinedStories = variants.length ? combineStories(defaultStory, variants) : defaultStory;
+  const defaultParams = {
```
Comment (kevinbuhmann) [inline .storybook/helpers/setup-storybook.helpers.ts:39]: I think this name is too similar to `defaultParameters`, so I would rename it `defaultVariantParameters`.

File: .storybook/helpers/setup-storybook.helpers.ts [hunk 2]
```diff
@@ -32,13 +34,31 @@ export function setupStorybook(
...
+    ...defaultParams,
+    themeOverride: THEMES.CORE_LIGHT,
+  });
+  storyApi.add('Variants - Core Dark Theme', combinedStories, {
+    ...defaultParams,
+    themeOverride: THEMES.CORE_DARK,
+  });
+  storyApi.add('Variants - High Contrast Theme', combinedStories, {
+    ...defaultParams,
+    themeOverride: THEMES.HIGH_CONTRAST,
```
Comment (kevinbuhmann) [inline .storybook/helpers/setup-storybook.helpers.ts]: I think a loop would be better here:

```typescript
for (const themeKey of Object.keys(THEMES)) {
    storyApi.add(`Variants - ${themeKey}`, combinedStories, {
      ...defaultVariantParameters,
      themeOverride: THEMES[themeKey],
    });
  }
  ```

File: .storybook/helpers/setup-storybook.helpers.ts [hunk 3]
```diff
@@ -44,22 +45,24 @@ export function setupStorybook(
...
+    });
+    storyApi.add('Variants - Core Dark Theme', combinedStories, {
+      ...defaultParams,
+      themeOverride: THEMES.CORE_DARK,
+    });
+    storyApi.add('Variants - High Contrast Theme', combinedStories, {
+      ...defaultParams,
+      themeOverride: THEMES.HIGH_CONTRAST,
+    });
+  }
```
Comment (kevinbuhmann) [inline .storybook/helpers/setup-storybook.helpers.ts:65]: Oh. This kinda conflicts with my request to use a loop. But we could do this:

```typescript
if (isChromatic()) {
  // loop from other comment
} else {
  storyApi.add('Variants', combinedStories, defaultVariantParameters);
}
```

File: .storybook/public/manager.css [hunk 4]
```diff
@@ -4,6 +4,6 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-a[id*='--variants'] {
+a[id$='--variants'] {
```
Comment (kevinbuhmann) [inline .storybook/public/manager.css:7]: Maybe this commit should be squashed into the previous one so we don't change this line and then change it back?

**Review by kevinbuhmann — DISMISSED** [review]
minor comments

File: .storybook/preview.js [hunk 5]
```diff
@@ -42,33 +43,36 @@ export const globalTypes = {
...
 };
 
-const themeDecorator = (story, { globals }) => {
+const themeDecorator = (story, { globals, parameters }) => {
   const { theme } = globals;
+  const { themeOverride } = parameters;
 
-  switch (theme) {
-    case '':
+  const currentTheme = themeOverride || theme;
```
Comment (kevinbuhmann) [inline .storybook/preview.js:60]: This a really clever solution! Thanks!

File: .storybook/helpers/setup-storybook.helpers.ts [hunk 6]
```diff
@@ -32,14 +35,27 @@ export function setupStorybook(
...
+
+  if (isChromatic()) {
+    for (const [themeKey, theme] of Object.values(THEMES)) {
+      storyApi.add(`Variants - ${themeKey}`, combinedStories, {
+        ...defaultVariantParameters,
+        themeOverride: theme,
+      });
+    }
+  } else {
+    storyApi.add('Variants', combinedStories, defaultVariantParameters);
```
Comment (kevinbuhmann) [inline .storybook/helpers/setup-storybook.helpers.ts:57]: This is duplicated now.

File: .github/workflows/pr-visual-regression.yml [hunk 7]
```diff
@@ -50,4 +50,4 @@ jobs:
           CHROMATIC_SHA: ${{steps.get_pr_event.outputs.sourceHeadSha}}
           CHROMATIC_BRANCH: ${{steps.get_pr_event.outputs.sourceHeadBranch}}
           CHROMATIC_SLUG: ${{steps.get_pr_event.outputs.sourceHeadRepo}}
-        run: npx chromatic --storybook-build-dir ./dist/docs --exit-once-uploaded
+        run: npx chromatic --only-changed --storybook-build-dir ./dist/docs --exit-once-uploaded
```
Comment (kevinbuhmann) [inline .github/workflows/pr-visual-regression.yml:53]: There is an "auto accept" workflow that might need this flag too.

Comment (williamernest) [reply .github/workflows/pr-visual-regression.yml:53]: Good catch

**Review by kevinbuhmann — APPROVED** [review]

**Pull request merged**